### PR TITLE
Upgrade GDP Master to yocto 2.2 (morty)

### PR DESCRIPTION
**Pull request by chbae**
[GDP-550] Upgrade GDP Master to yocto 2.2 (morty)

### CODE REVIEW

**Comment by chbae** [comment]
This PR should be tested with https://github.com/GENIVI/meta-genivi-dev/pull/74

**Comment by chbae** [comment]
I fixed dragonboard problem. Now PR is working on all devices (qemux86, raspberrypi3, minnowboard, dragonboard). We need to check renesas board.

@slawr FYI

**Comment by chbae** [comment]
> Why are we updating m-g-d submodule multiple times in this PR? I think we should have to do it just once.
I'll do it.

File: gdp-src-build/conf/templates/raspberrypi.local.inc [hunk 1]
```diff
@@ -2,19 +2,6 @@ include local.inc
 
 GPU_MEM = "128"
 
-CORE_IMAGE_EXTRA_INSTALL += "wayland weston"
```
Comment (gunnarx) [inline gdp-src-build/conf/templates/raspberrypi.local.inc]: I love all the lines that can be deleted here :-)

**Comment by chbae** [comment]
Rebased and use just one commit as @zeenix 's comment.

**Comment by zeenix** [comment]
@chbae When I try to update submodules your branch, I get an error:

```
$ git submodule update 
Submodule path 'meta-intel': checked out '6add41510412ca196efb3e4f949d403a8b6f35d7'
error: no such remote ref cee97c5335a34d97d0940728254c32fa7f773519
Fetched in submodule path 'meta-ivi', but it did not contain cee97c5335a34d97d0940728254c32fa7f773519. Direct fetching of that commit failed.
```

**Comment by chbae** [comment]
@zeenix Then 4 commits (qemux86, raspberrypi, minnowboard, dragonboard) are fine to me.

**Comment by gunnarx** [comment]
Both me and @zeenix seem to have been fooled, because cee97c533 is not on meta-ivi 12.0 branch!
As chbae showed, N-0.2 tag is however equal to cee97c533:

```
$ git log N-0.2 --not 12.0
commit cee97c5335a34d97d0940728254c32fa7f773519 (tag: N-0.2)
Author: Yong-iL Joh <yong-il.joh@windriver.com>
Date:   Thu Mar 9 17:35:32 2017 +0900

    dbus: remove "not essential patches for common-api"
    
    Signed-off-by: Yong-iL Joh <yong-il.joh@windriver.com>
```

I had only performed a git-fetch in meta-ivi and when i did not find the commit I assumed it was not there.  And zeenix's submodule update I suppose also fetched all branches but not a tag that is detached from the branches?

Of course 
```
git fetch --tags
``` 
will remedy the situation.

So, some other branch (anonymous, possibly detached HEAD) seems to have been used by @tolkien-joh when committing and setting the tag - it is not (yet?) on the 12.0 development branch.  What do you think Tolkien?

**Comment by gunnarx** [comment]
Build checks: I think Koelsch/Porter is not set up to support Morty, so we'll see what to do about that later.  

Other failures might simply be that it needs https://github.com/GENIVI/meta-genivi-dev/pull/74 to be merged also, as we all know.

Anyhow, because of the complexity of this PR, me and @zeenix eventually agreed to merge this into a separate morty branch for a short while.  I have already set up the separate build pipelines. That way it can be worked on separately and the build checks will not be interfered by other PRs.  And there are some new hardware targets that should be confirmed to be ok also before this goes in to master.

**Comment by chbae** [comment]
@gunnarx I agree with you and I'm waiting for @tolkien-joh's change about N-0.2 tag.

**Comment by gunnarx** [comment]
It seems to me that the same patch as cee97c533 exists 2 steps earlier on the 12.0 branch with hash 860f0ce454:

Comparing the diff 2 steps down from 12.0 tip.
```
13:31 $ git diff 12.0~3 12.0~2
diff --git a/meta-ivi/recipes-core-ivi/dbus/dbus_%.bbappend b/meta-ivi/recipes-core-ivi/dbus/dbus_%.bbappend
index 407535b..2b2446e 100644
--- a/meta-ivi/recipes-core-ivi/dbus/dbus_%.bbappend
+++ b/meta-ivi/recipes-core-ivi/dbus/dbus_%.bbappend
@@ -5,8 +5,5 @@ FILESEXTRAPATHS_append := ":${THISDIR}/${PN}"
 SRC_URI_append = " \
     file://capi-dbus-add-send-with-reply-set-notify.patch \
     file://capi-dbus-add-support-for-custom-marshalling.patch \
-    file://capi-dbus-block-acquire-io-path-on-send.patch \
-    file://capi-dbus-correct-dbus-connection-block-pending-call.patch \
-    file://capi-dbus-send-with-reply-and-block-delete-reply-on-error.patch \
     "
```

and: N-0.2 = cee97c533:

```
$ git diff N-0.2^ N-0.2
diff --git a/meta-ivi/recipes-core-ivi/dbus/dbus_%.bbappend b/meta-ivi/recipes-core-ivi/dbus/dbus_%.bbappend
index 407535b..2b2446e 100644
--- a/meta-ivi/recipes-core-ivi/dbus/dbus_%.bbappend
+++ b/meta-ivi/recipes-core-ivi/dbus/dbus_%.bbappend
@@ -5,8 +5,5 @@ FILESEXTRAPATHS_append := ":${THISDIR}/${PN}"
 SRC_URI_append = " \
     file://capi-dbus-add-send-with-reply-set-notify.patch \
     file://capi-dbus-add-support-for-custom-marshalling.patch \
-    file://capi-dbus-block-acquire-io-path-on-send.patch \
-    file://capi-dbus-correct-dbus-connection-block-pending-call.patch \
-    file://capi-dbus-send-with-reply-and-block-delete-reply-on-error.patch \
     "
 SRC_URI_append_qemux86 = " file://build_fix-ubunt1604-qemux86.patch"
```

but the dlt patch that follows it is currently failing in CI, so I am unsure if the tip of 12.0 could work as an alternative to N-0.2.   So how about we just pick 860f0ce454 for now and get moving on testing this?

**Comment by gunnarx** [comment]
Pushed one more commit ^^ that uses meta-ivi 860f0ce454 instead. 
@chbae if you want, we could squash my commit 68efc51 into your 40dd152 for a cleaner history.

I also pushed this to genivi/morty now for easier testing.  When we are happy with the result we could sync up this PR with that branch and merge the PR into master (for GitHub documentation purposes).

**Comment by chbae** [comment]
@gunnarx I rebased and squashed your commit in both my repo and morty branch.
Thanks.

**Comment by gunnarx** [comment]
OK.  So far so good.  We won't support Koelsch and Porter in the first run because the BSP is not up to date for this yet.   The supported hardware builds are OK except for Minnowboard for unknown reason - investigating.

**Comment by chbae** [comment]
Could you let me know about Minnowboard error?

**Comment by gunnarx** [comment]
Build checks: I think Koelsch/Porter is not set up to support Morty, so we'll see what to do about that later.  

Anyhow, because of the complexity of this PR, me and @zeenix eventually agreed to merge this into a separate morty branch for a short while.  I have set up separate build pipelines for it. That way it can be worked on separately and the build checks will not be interfered by other PRs. 

We're working through separate issues for some other failures.

**Comment by gunnarx** [comment]
@chbae 
>Could you let me know about Minnowboard error?

According to my investigations last week of 79002aeb7d (tip of morty branch) on multiple builds on different machines - this seems to succeed every time on a fully clean build but fail if it's built again with reused sstate?  So something is not as it should be.
I had one different failure from time to time but most of the time it complains about mesa/open-gl files which suggests looking at the particular settings done in the conf file.
Example: https://go.genivi.org/go/tab/build/detail/Morty-GDP-Yocto-Minnowboard/5/build/1/init_and_bitbake

**Comment by gunnarx** [comment]
@chbae As an alternative I am also building this without the special settings in the Minnowboard conf file.  (i.e. with a totally standard Intel BSP).  That builds successfully, also on repeated builds with sstate.  

I'd like to think we try the image result from that and see.  If the image runs correctly, then the special settings are nowadays obsolete and we can build with a vanilla BSP instead.

**Comment by chbae** [comment]
I'll rebase and apply N-1.0 of meta-ivi

**Comment by slawr** [comment]
Really just thinking aloud but working on Morty I see we didn't get around to either moving to Qt 5.6.3 or back porting the upstream fix (in Qt 5.6.3) to aarch64 pointer handling in QtQml. It means aarch64 yocto BSPs will need to continue to carry workarounds.

**Comment by chbae** [comment]
@slawr I'll upgrade qt to 5.6.3 after merging morty branch.

**Comment by zeenix** [comment]
@chbae But Qt 5.6.3 is not even released: http://download.qt.io/official_releases/qt/5.6/

**Comment by zeenix** [comment]
@slawr this is about https://at.projects.genivi.org/jira/browse/GDP-549 right? If so, my understanding was that we don't yet have the patches ported. Is that not correct?

**Comment by slawr** [comment]
@chbae that would be great. Although looking at the links provided in GDP-549 it looks like their plans for the 5.6.3 release have not firmed up. We may need to go back to the Qt maintainers to see what is happening. We should probably take detailed discussion of that over to the GDP-549 ticket though to avoid polluting this PR thread too much. As I said I was just thinking aloud :)

@zeenix yes GDP-549. I don't know the status. Seems like upstream needs chasing.

**Comment by gunnarx** [comment]
Commenting on Build checks: 
(14:23:11.177 ERROR: ParseError at.../python-pyqt_5.3.1.bb:28: Could not inherit file classes/sip.bbclass)
Just as before:
It is because PRs are tested against master and not morty.  This update is checked on separate pipelines for Morty instead and I (or someone else) will report on it separately.

**Comment by zeenix** [comment]
@gunnarx Thanks for reminding. We'll have to test locally then. This would be another reason to try to merge morty first i guess.

**Comment by zeenix** [comment]
@chbae Just so you know, I've rebased the morty branch on latest git master. You might want to update your branch/clone.

**Comment by gunnarx** [comment]
Same as before, c850555c, (this PR rebased on latest master, and the same for meta-genivi-dev PR), builds OK for R-Pi, Dragonboard.  Intel minnowboard builds fine clean but not with reused sstate.  (Have a pretty good idea what needs to change in the conf to fix that though).

**Comment by chbae** [comment]
@gunnarx Please re-trigger again for minnowboard

**Comment by zeenix** [comment]
So rebased the morty branch in mainline repo to:

* Rebase against current git master.
* Address the issue I pointed out in https://github.com/GENIVI/genivi-dev-platform/pull/67#pullrequestreview-27815530 . i-e I squashed all changes to g-d-p repo into 1 single commit.

**Pull request merged**